model/src/main/java/org/projectnessie/model/Namespace.java

```diff
@@ -96,7 +97,7 @@ public static Namespace of(String... elements) {
             String.format(
                 "Namespace '%s' must not contain a null element.", Arrays.toString(elements)));
       }
-      if (e.contains(ZERO_BYTE_STRING)) {
+      if (e.contains(ZERO_BYTE_STRING) || e.contains(ZERO_BYTE_STRING_URL_ENCODED)) {
```

Comment:
The second check seems unnecessary to me. `URLEncoder` should be able to handle it in `toPathString`.

Reply:
without this check something like `Namespace.of("%00")` will not fail, but `Namespace.of("\u0000") will. This is because validation happens before url encoding

Reply:
true, but I think it would be correct.